### PR TITLE
[flutter_runner] a11y updates, tests!

### PR DESCRIPTION
**Pull request by dnfield**
This changes the tests from topaz since they didn't all make sense anymore.

/cc @ankitdave06 for help on reviewing the a11y changes in here.

You can see a sample run of the tests here (requires login):
https://chromium-swarm.appspot.com/task?id=4769b2b476457510

After this commit, we will be running _actual_ tests on Fuchsia with each commit.

I've modified the platform view tests - semantics manager is not in the SDK and is going away entirely in favor of new methods on SemanticListener.  Because of that, it no longer makes sense to test the platform view for semantics manager related things (But I do still test that it sets features as expected).  I've added a test to AccessibilityBridgeTests to verify that it notifies the delegate (e.g. platform view) of changes to status.

### CODE REVIEW

**Comment by dnfield** [comment]
Test output for Fuchsia from this try run: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8901755467113849248/+/steps/collect/0/logs/task_stdout_stderr:_flutter_runner_fuchsia/0

**Comment by dnfield** [comment]
This isn't actually passing - CI is wrong.  Working to fix CI first and then will try to fix the failing test.

File: shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  }
+
+  void RunLoopUntilIdle() {
+    loop_.RunUntilIdle();
+    loop_.ResetQuit();
+  }
+
+ protected:
+  void SetUp() override {
+    zx::eventpair a, b;
```
Comment (ankitdave06) [inline shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc]: Please store these variables as class member and make sure they don't go out of scope.
I have sent an email regarding this.

I am about to land a change which will add a event listener which will delete the semantic tree and close fidl channel with semantics manager when eventpairs go out of scope.

Comment (ankitdave06) [reply shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc]: Also consider changing their name.

File: shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc [hunk 2]
```diff
@@ -0,0 +1,297 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
```
Comment (ankitdave06) [inline shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc]: May be we should also consider adding more test cases for sending partial updates.
Not sure if accessibility bridges supports it at this moment. But something to remember in the future.

File: shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+  AccessibilityBridgeTestDelegate accessibility_delegate_;
+  std::unique_ptr<flutter_runner::AccessibilityBridge> accessibility_bridge_;
+
+ private:
+  async::Loop loop_;
+  sys::testing::ServiceDirectoryProvider services_provider_;
+};
+
+TEST_F(AccessibilityBridgeTest, RegistersViewRef) {
+  EXPECT_TRUE(semantics_manager_.RegisterViewCalled());
```
Comment (ankitdave06) [inline shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc]: These function name should change to RegisterViewForSemantics() ?

**Comment by ankitdave06** [comment]
Mostly looks good. I added few comments based on the recent changes that happened and will happen on semantics manager side.

File: shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc (hunk 1 above)
Comment (dnfield) [reply shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc]: Do they have to stay in scope, or the clone we make for a11y, or both?

File: shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc (hunk 2 above)
Comment (dnfield) [reply shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc]: That's a good idea, but probably for a separate PR

File: shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc (hunk 3 above)
Comment (dnfield) [reply shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc]: This is a method on the fake to check whether `RegisterViewForSemantics` was called or not.

File: shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc (hunk 1 above)
Comment (dnfield) [reply shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc]: I'll make the change in the test and in the implementation.  I see now what you're saying - the clone is of a different object and may not matter.

Comment (dnfield) [reply shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc]: Implementation was fine - it's held by the platform view.  Updated the test.

File: shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc (hunk 2 above)
Comment (ankitdave06) [reply shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc]: sounds good.

File: shell/platform/fuchsia/flutter/platform_view_unittest.cc [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+  EXPECT_EQ(delegate.SemanticsFeatures(), 0);
+}
+
+// Test to make sure that PlatformView correctly registers messages sent on
+// the "flutter/platform_views" channel, correctly parses the JSON it receives
+// and calls the EnableWireframeCallback with the appropriate args.
+TEST_F(PlatformViewTests, EnableWireframeTest) {
+  sys::testing::ServiceDirectoryProvider services_provider(dispatcher());
+  MockPlatformViewDelegate delegate;
+  zx::eventpair a, b;
```
Comment (ankitdave06) [inline shell/platform/fuchsia/flutter/platform_view_unittest.cc]: Why do we need this?
SetUp() is already creating eventpair using viewrefs
This looks unnecessary

Comment (dnfield) [reply shell/platform/fuchsia/flutter/platform_view_unittest.cc]: That's for a different test... the a11y tests are structured differently.

Comment (ankitdave06) [reply shell/platform/fuchsia/flutter/platform_view_unittest.cc]: Sorry I mixed them. Not very used to this ui.

Comment (ankitdave06) [reply shell/platform/fuchsia/flutter/platform_view_unittest.cc]: [Optional] I would recommend changing names of a and b here to something more meaningful.

**Pull request merged**